// ng2-owl-carousel/src/owl-carousel.component.ngfactory.js
"use strict";
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("./owl-child.component.ngfactory");
var i2 = require("@angular/common");
var i3 = require("ng2-owl-carousel/src/owl-child.component");
var i4 = require("ng2-owl-carousel/src/owl-carousel.component");
var styles_OwlCarousel = [];
exports.RenderType_OwlCarousel = i0.ɵcrt({ encapsulation: 2, styles: styles_OwlCarousel,
    data: {} });
function View_OwlCarousel_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 3, 'owl-carousel-child', [], [[2, 'owl-carousel', null]], null, null, i1.View_OwlChild_0, i1.RenderType_OwlChild)), i0.ɵdid(1, 278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngClass: [0, 'ngClass'] }, null), i0.ɵdid(2, 4374528, [[1, 4], ['owl', 4]], 0, i3.OwlChild, [i0.ElementRef], { options: [0, 'options'] }, null), i0.ɵncd(0, 0)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _co.carouselClasses;
        _ck(_v, 1, 0, currVal_1);
        var currVal_2 = _co.options;
        _ck(_v, 2, 0, currVal_2);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 2).owlClass;
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_OwlCarousel_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(671088640, 1, { $owlChild: 0 }), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OwlCarousel_1)), i0.ɵdid(2, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.show;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
exports.View_OwlCarousel_0 = View_OwlCarousel_0;
function View_OwlCarousel_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'owl-carousel', [], null, null, null, View_OwlCarousel_0, exports.RenderType_OwlCarousel)),
        i0.ɵdid(1, 311296, null, 0, i4.OwlCarousel, [i0.IterableDiffers], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_OwlCarousel_Host_0 = View_OwlCarousel_Host_0;
exports.OwlCarouselNgFactory = i0.ɵccf('owl-carousel', i4.OwlCarousel, View_OwlCarousel_Host_0, { carouselClasses: 'carouselClasses', options: 'options',
    items: 'items' }, {}, ['*']);
