// angular4-notify/angular4-notify.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'angular4-notify';
import * as i2 from '@angular/common';
import * as i3 from '@angular/router';
export const NotificationsModuleNgFactory:i0.NgModuleFactory<i1.NotificationsModule> = i0.ɵcmf(i1.NotificationsModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i1.NotificationsModule,i1.NotificationsModule,
          ([] as any[]))]);
    });
const styles_ContainerComponent:any[] = ['.notifications-container[_ngcontent-%COMP%] { width: 100%; } .notify[_ngcontent-%COMP%] { width: 100%; padding: 8px 16px; } .notify[_ngcontent-%COMP%]   i[_ngcontent-%COMP%] { margin-right: 16px; vertical-align: middle; cursor: pointer; } .error[_ngcontent-%COMP%] { background-color: #ef4a4a; } .info[_ngcontent-%COMP%] { background-color: #60db8f; } .warning[_ngcontent-%COMP%] { background-color: #dbb460; }'];
export const RenderType_ContainerComponent:i0.RendererType2 = i0.ɵcrt({encapsulation:0,
    styles:styles_ContainerComponent,data:{}});
function View_ContainerComponent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'angular4-notify-notification',
      ([] as any[]),(null as any),[[(null as any),'dismissNotify']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('dismissNotify' === en)) {
          const pd_0:any = ((<any>_co.onDismissNotify($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_NotificationComponent_0,RenderType_NotificationComponent)),i0.ɵdid(1,
      49152,(null as any),0,i1.NotificationComponent,([] as any[]),{notification:[0,
          'notification']},{dismissNotify:'dismissNotify'})],(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit;
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
export function View_ContainerComponent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),4,'div',[['class',
      'notifications-container']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted(-1,(null as any),[' '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_ContainerComponent_1)),i0.ɵdid(3,
      802816,(null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
      {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),[' ']))],
      (_ck,_v) => {
        var _co:i1.ContainerComponent = _v.component;
        const currVal_0:any = _co.notifications;
        _ck(_v,3,0,currVal_0);
      },(null as any));
}
export function View_ContainerComponent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'angular4-notify-notifications-container',
      ([] as any[]),(null as any),(null as any),(null as any),View_ContainerComponent_0,
      RenderType_ContainerComponent)),i0.ɵdid(1,245760,(null as any),0,i1.ContainerComponent,
      [i1.NotificationsService,i3.Router],(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const ContainerComponentNgFactory:i0.ComponentFactory<i1.ContainerComponent> = i0.ɵccf('angular4-notify-notifications-container',
    i1.ContainerComponent,View_ContainerComponent_Host_0,{},{},([] as any[]));
const styles_NotificationComponent:any[] = ['.notifications-container[_ngcontent-%COMP%] { width: 100%; } .notify[_ngcontent-%COMP%] { width: 100%; padding: 8px 16px; } .notify[_ngcontent-%COMP%]   i[_ngcontent-%COMP%] { margin-right: 16px; vertical-align: middle; cursor: pointer; } .error[_ngcontent-%COMP%] { background-color: #ef4a4a; } .info[_ngcontent-%COMP%] { background-color: #60db8f; } .warning[_ngcontent-%COMP%] { background-color: #dbb460; }'];
export const RenderType_NotificationComponent:i0.RendererType2 = i0.ɵcrt({encapsulation:0,
    styles:styles_NotificationComponent,data:{}});
export function View_NotificationComponent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),7,'div',([] as any[]),
      [[8,'className',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted(-1,(null as any),[' '])),(_l()(),i0.ɵeld(2,0,(null as any),(null as any),
          4,'a',[['class','close'],['type','button']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted(-1,(null as any),
          [' '])),(_l()(),i0.ɵeld(4,0,(null as any),(null as any),1,'i',[['class',
          'material-icons']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.NotificationComponent = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.dismiss()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted(-1,(null as any),['clear'])),
      (_l()(),i0.ɵted(-1,(null as any),[' '])),(_l()(),i0.ɵted(7,(null as any),[' ',
          ' '])),(_l()(),i0.ɵted(-1,(null as any),[' ']))],(null as any),(_ck,_v) => {
    var _co:i1.NotificationComponent = _v.component;
    const currVal_0:any = i0.ɵinlineInterpolate(1,'notify ',_co.notification.type,
        '');
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = _co.notification.message;
    _ck(_v,7,0,currVal_1);
  });
}
export function View_NotificationComponent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'angular4-notify-notification',
      ([] as any[]),(null as any),(null as any),(null as any),View_NotificationComponent_0,
      RenderType_NotificationComponent)),i0.ɵdid(1,49152,(null as any),0,i1.NotificationComponent,
      ([] as any[]),(null as any),(null as any))],(null as any),(null as any));
}
export const NotificationComponentNgFactory:i0.ComponentFactory<i1.NotificationComponent> = i0.ɵccf('angular4-notify-notification',
    i1.NotificationComponent,View_NotificationComponent_Host_0,{notification:'notification'},
    {dismissNotify:'dismissNotify'},([] as any[]));
