// ngx-pagination/dist/ngx-pagination.ngfactory.js
"use strict";
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("ngx-pagination");
var i2 = require("@angular/common");
exports.NgxPaginationModuleNgFactory = i0.ɵcmf(i1.NgxPaginationModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(4608, i1.PaginationService, i1.PaginationService, []), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.NgxPaginationModule, i1.NgxPaginationModule, [])]);
});
var styles_PaginationControlsComponent = ['\n.ngx-pagination {\n  margin-left: 0;\n  margin-bottom: 1rem; }\n  .ngx-pagination::before, .ngx-pagination::after {\n    content: \' \';\n    display: table; }\n  .ngx-pagination::after {\n    clear: both; }\n  .ngx-pagination li {\n    -moz-user-select: none;\n    -webkit-user-select: none;\n    -ms-user-select: none;\n    margin-right: 0.0625rem;\n    border-radius: 0; }\n  .ngx-pagination li {\n    display: inline-block; }\n  .ngx-pagination a,\n  .ngx-pagination button {\n    color: #0a0a0a; \n    display: block;\n    padding: 0.1875rem 0.625rem;\n    border-radius: 0; }\n    .ngx-pagination a:hover,\n    .ngx-pagination button:hover {\n      background: #e6e6e6; }\n  .ngx-pagination .current {\n    padding: 0.1875rem 0.625rem;\n    background: #2199e8;\n    color: #fefefe;\n    cursor: default; }\n  .ngx-pagination .disabled {\n    padding: 0.1875rem 0.625rem;\n    color: #cacaca;\n    cursor: default; } \n    .ngx-pagination .disabled:hover {\n      background: transparent; }\n  .ngx-pagination .ellipsis::after {\n    content: \'…\';\n    padding: 0.1875rem 0.625rem;\n    color: #0a0a0a; }\n  .ngx-pagination a, .ngx-pagination button {\n    cursor: pointer; }\n\n.ngx-pagination .pagination-previous a::before,\n.ngx-pagination .pagination-previous.disabled::before { \n  content: \'«\';\n  display: inline-block;\n  margin-right: 0.5rem; }\n\n.ngx-pagination .pagination-next a::after,\n.ngx-pagination .pagination-next.disabled::after {\n  content: \'»\';\n  display: inline-block;\n  margin-left: 0.5rem; }\n\n.ngx-pagination .show-for-sr {\n  position: absolute !important;\n  width: 1px;\n  height: 1px;\n  overflow: hidden;\n  clip: rect(0, 0, 0, 0); }'];
exports.RenderType_PaginationControlsComponent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_PaginationControlsComponent, data: {} });
function View_PaginationControlsComponent_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 4, 'a', [['tabindex',
                '0']], [[1, 'aria-label', 0]], [[null, 'keyup.enter'], [null, 'click']], function (_v, en, $event) {
            var ad = true;
            if (('keyup.enter' === en)) {
                var pd_0 = (i0.ɵnov(_v.parent.parent.parent, 2).previous() !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (i0.ɵnov(_v.parent.parent.parent, 2).previous() !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(1, null, ['\n                ',
            ' '])), (_l()(), i0.ɵeld(2, 0, null, null, 1, 'span', [['class', 'show-for-sr']], null, null, null, null, null)), (_l()(),
            i0.ɵted(3, null, ['', ''])), (_l()(), i0.ɵted(-1, null, ['\n            ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = ((_co.previousLabel + ' ') + _co.screenReaderPageLabel);
        _ck(_v, 0, 0, currVal_0);
        var currVal_1 = _co.previousLabel;
        _ck(_v, 1, 0, currVal_1);
        var currVal_2 = _co.screenReaderPageLabel;
        _ck(_v, 3, 0, currVal_2);
    });
}
function View_PaginationControlsComponent_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 4, 'span', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(1, null, ['\n                ', ' '])), (_l()(), i0.ɵeld(2, 0, null, null, 1, 'span', [['class', 'show-for-sr']], null, null, null, null, null)), (_l()(), i0.ɵted(3, null, ['', ''])), (_l()(), i0.ɵted(-1, null, ['\n            ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.previousLabel;
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = _co.screenReaderPageLabel;
        _ck(_v, 3, 0, currVal_1);
    });
}
function View_PaginationControlsComponent_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 7, 'li', [['class',
                'pagination-previous']], [[2, 'disabled', null]], null, null, null, null)), (_l()(), i0.ɵted(-1, null, [' \n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_3)),
        i0.ɵdid(3, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_4)),
        i0.ɵdid(6, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n        ']))], function (_ck, _v) {
        var currVal_1 = (1 < i0.ɵnov(_v.parent.parent, 2).getCurrent());
        _ck(_v, 3, 0, currVal_1);
        var currVal_2 = i0.ɵnov(_v.parent.parent, 2).isFirstPage();
        _ck(_v, 6, 0, currVal_2);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v.parent.parent, 2).isFirstPage();
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_PaginationControlsComponent_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 7, 'a', [['tabindex',
                '0']], null, [[null, 'keyup.enter'], [null, 'click']], function (_v, en, $event) {
            var ad = true;
            if (('keyup.enter' === en)) {
                var pd_0 = (i0.ɵnov(_v.parent.parent.parent, 2).setCurrent(_v.parent.context.$implicit.value) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (i0.ɵnov(_v.parent.parent.parent, 2).setCurrent(_v.parent.context.$implicit.value) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['\n                '])),
        (_l()(), i0.ɵeld(2, 0, null, null, 1, 'span', [['class', 'show-for-sr']], null, null, null, null, null)),
        (_l()(), i0.ɵted(3, null, ['', ' '])), (_l()(), i0.ɵted(-1, null, ['\n                '])), (_l()(), i0.ɵeld(5, 0, null, null, 1, 'span', [], null, null, null, null, null)), (_l()(), i0.ɵted(6, null, ['', ''])), (_l()(), i0.ɵted(-1, null, ['\n            ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.screenReaderPageLabel;
        _ck(_v, 3, 0, currVal_0);
        var currVal_1 = _v.parent.context.$implicit.label;
        _ck(_v, 6, 0, currVal_1);
    });
}
function View_PaginationControlsComponent_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 7, null, null, null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n                '])), (_l()(), i0.ɵeld(2, 0, null, null, 1, 'span', [['class', 'show-for-sr']], null, null, null, null, null)), (_l()(), i0.ɵted(3, null, ['', ' '])), (_l()(), i0.ɵted(-1, null, ['\n                '])),
        (_l()(), i0.ɵeld(5, 0, null, null, 1, 'span', [], null, null, null, null, null)), (_l()(), i0.ɵted(6, null, ['', ''])), (_l()(), i0.ɵted(-1, null, [' \n            ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.screenReaderCurrentLabel;
        _ck(_v, 3, 0, currVal_0);
        var currVal_1 = _v.parent.context.$implicit.label;
        _ck(_v, 6, 0, currVal_1);
    });
}
function View_PaginationControlsComponent_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 7, 'li', [], [[2, 'current', null]], null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_6)),
        i0.ɵdid(3, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_7)),
        i0.ɵdid(6, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n        ']))], function (_ck, _v) {
        var currVal_1 = (i0.ɵnov(_v.parent.parent, 2).getCurrent() !== _v.context.$implicit.value);
        _ck(_v, 3, 0, currVal_1);
        var currVal_2 = (i0.ɵnov(_v.parent.parent, 2).getCurrent() === _v.context.$implicit.value);
        _ck(_v, 6, 0, currVal_2);
    }, function (_ck, _v) {
        var currVal_0 = (i0.ɵnov(_v.parent.parent, 2).getCurrent() === _v.context.$implicit.value);
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_PaginationControlsComponent_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 4, 'a', [['tabindex',
                '0']], [[1, 'aria-label', 0]], [[null, 'keyup.enter'], [null, 'click']], function (_v, en, $event) {
            var ad = true;
            if (('keyup.enter' === en)) {
                var pd_0 = (i0.ɵnov(_v.parent.parent.parent, 2).next() !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (i0.ɵnov(_v.parent.parent.parent, 2).next() !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(1, null, ['\n                 ',
            ' '])), (_l()(), i0.ɵeld(2, 0, null, null, 1, 'span', [['class', 'show-for-sr']], null, null, null, null, null)), (_l()(),
            i0.ɵted(3, null, ['', ''])), (_l()(), i0.ɵted(-1, null, ['\n            ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = ((_co.nextLabel + ' ') + _co.screenReaderPageLabel);
        _ck(_v, 0, 0, currVal_0);
        var currVal_1 = _co.nextLabel;
        _ck(_v, 1, 0, currVal_1);
        var currVal_2 = _co.screenReaderPageLabel;
        _ck(_v, 3, 0, currVal_2);
    });
}
function View_PaginationControlsComponent_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 4, 'span', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(1, null, ['\n                 ', ' '])), (_l()(), i0.ɵeld(2, 0, null, null, 1, 'span', [['class', 'show-for-sr']], null, null, null, null, null)), (_l()(), i0.ɵted(3, null, ['', ''])), (_l()(), i0.ɵted(-1, null, ['\n            ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.nextLabel;
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = _co.screenReaderPageLabel;
        _ck(_v, 3, 0, currVal_1);
    });
}
function View_PaginationControlsComponent_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 7, 'li', [['class',
                'pagination-next']], [[2, 'disabled', null]], null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_9)),
        i0.ɵdid(3, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_10)),
        i0.ɵdid(6, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n        ']))], function (_ck, _v) {
        var currVal_1 = !i0.ɵnov(_v.parent.parent, 2).isLastPage();
        _ck(_v, 3, 0, currVal_1);
        var currVal_2 = i0.ɵnov(_v.parent.parent, 2).isLastPage();
        _ck(_v, 6, 0, currVal_2);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v.parent.parent, 2).isLastPage();
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_PaginationControlsComponent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 10, 'ul', [['class',
                'ngx-pagination'], ['role', 'navigation']], [[1, 'aria-label', 0]], null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_2)), i0.ɵdid(3, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(-1, null, ['\n\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_5)), i0.ɵdid(6, 802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n\n        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_8)),
        i0.ɵdid(9, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _co.directionLinks;
        _ck(_v, 3, 0, currVal_1);
        var currVal_2 = i0.ɵnov(_v.parent, 2).pages;
        _ck(_v, 6, 0, currVal_2);
        var currVal_3 = _co.directionLinks;
        _ck(_v, 9, 0, currVal_3);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.screenReaderPaginationLabel;
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_PaginationControlsComponent_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵeld(1, 0, null, null, 5, 'pagination-template', [], null, [[null, 'pageChange']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('pageChange' === en)) {
                var pd_0 = (_co.pageChange.emit($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(2, 737280, [['p', 4]], 0, i1.PaginationControlsDirective, [i1.PaginationService, i0.ChangeDetectorRef], { id: [0, 'id'], maxSize: [1, 'maxSize'] }, { pageChange: 'pageChange' }), (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_PaginationControlsComponent_1)),
        i0.ɵdid(5, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n    '])),
        (_l()(), i0.ɵted(-1, null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.id;
        var currVal_1 = _co.maxSize;
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = !(_co.autoHide && (i0.ɵnov(_v, 2).pages.length <= 1));
        _ck(_v, 5, 0, currVal_2);
    }, null);
}
exports.View_PaginationControlsComponent_0 = View_PaginationControlsComponent_0;
function View_PaginationControlsComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'pagination-controls', [], null, null, null, View_PaginationControlsComponent_0, exports.RenderType_PaginationControlsComponent)), i0.ɵdid(1, 49152, null, 0, i1.PaginationControlsComponent, [], null, null)], null, null);
}
exports.View_PaginationControlsComponent_Host_0 = View_PaginationControlsComponent_Host_0;
exports.PaginationControlsComponentNgFactory = i0.ɵccf('pagination-controls', i1.PaginationControlsComponent, View_PaginationControlsComponent_Host_0, { id: 'id',
    maxSize: 'maxSize', directionLinks: 'directionLinks', autoHide: 'autoHide', previousLabel: 'previousLabel',
    nextLabel: 'nextLabel', screenReaderPaginationLabel: 'screenReaderPaginationLabel',
    screenReaderPageLabel: 'screenReaderPageLabel', screenReaderCurrentLabel: 'screenReaderCurrentLabel' }, { pageChange: 'pageChange' }, []);
