// angular4-notify/angular4-notify.ngfactory.js
"use strict";
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("angular4-notify");
var i2 = require("@angular/common");
var i3 = require("@angular/router");
exports.NotificationsModuleNgFactory = i0.ɵcmf(i1.NotificationsModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.NotificationsModule, i1.NotificationsModule, [])]);
});
var styles_ContainerComponent = ['.notifications-container[_ngcontent-%COMP%] { width: 100%; } .notify[_ngcontent-%COMP%] { width: 100%; padding: 8px 16px; } .notify[_ngcontent-%COMP%]   i[_ngcontent-%COMP%] { margin-right: 16px; vertical-align: middle; cursor: pointer; } .error[_ngcontent-%COMP%] { background-color: #ef4a4a; } .info[_ngcontent-%COMP%] { background-color: #60db8f; } .warning[_ngcontent-%COMP%] { background-color: #dbb460; }'];
exports.RenderType_ContainerComponent = i0.ɵcrt({ encapsulation: 0,
    styles: styles_ContainerComponent, data: {} });
function View_ContainerComponent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'angular4-notify-notification', [], null, [[null, 'dismissNotify']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('dismissNotify' === en)) {
                var pd_0 = (_co.onDismissNotify($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, View_NotificationComponent_0, exports.RenderType_NotificationComponent)), i0.ɵdid(1, 49152, null, 0, i1.NotificationComponent, [], { notification: [0,
                'notification'] }, { dismissNotify: 'dismissNotify' })], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit;
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
function View_ContainerComponent_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 4, 'div', [['class',
                'notifications-container']], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, [' '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContainerComponent_1)), i0.ɵdid(3, 802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(-1, null, [' ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.notifications;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
exports.View_ContainerComponent_0 = View_ContainerComponent_0;
function View_ContainerComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'angular4-notify-notifications-container', [], null, null, null, View_ContainerComponent_0, exports.RenderType_ContainerComponent)), i0.ɵdid(1, 245760, null, 0, i1.ContainerComponent, [i1.NotificationsService, i3.Router], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_ContainerComponent_Host_0 = View_ContainerComponent_Host_0;
exports.ContainerComponentNgFactory = i0.ɵccf('angular4-notify-notifications-container', i1.ContainerComponent, View_ContainerComponent_Host_0, {}, {}, []);
var styles_NotificationComponent = ['.notifications-container[_ngcontent-%COMP%] { width: 100%; } .notify[_ngcontent-%COMP%] { width: 100%; padding: 8px 16px; } .notify[_ngcontent-%COMP%]   i[_ngcontent-%COMP%] { margin-right: 16px; vertical-align: middle; cursor: pointer; } .error[_ngcontent-%COMP%] { background-color: #ef4a4a; } .info[_ngcontent-%COMP%] { background-color: #60db8f; } .warning[_ngcontent-%COMP%] { background-color: #dbb460; }'];
exports.RenderType_NotificationComponent = i0.ɵcrt({ encapsulation: 0,
    styles: styles_NotificationComponent, data: {} });
function View_NotificationComponent_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 7, 'div', [], [[8, 'className', 0]], null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, [' '])), (_l()(), i0.ɵeld(2, 0, null, null, 4, 'a', [['class', 'close'], ['type', 'button']], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, [' '])), (_l()(), i0.ɵeld(4, 0, null, null, 1, 'i', [['class',
                'material-icons']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.dismiss() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['clear'])),
        (_l()(), i0.ɵted(-1, null, [' '])), (_l()(), i0.ɵted(7, null, [' ',
            ' '])), (_l()(), i0.ɵted(-1, null, [' ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = i0.ɵinlineInterpolate(1, 'notify ', _co.notification.type, '');
        _ck(_v, 0, 0, currVal_0);
        var currVal_1 = _co.notification.message;
        _ck(_v, 7, 0, currVal_1);
    });
}
exports.View_NotificationComponent_0 = View_NotificationComponent_0;
function View_NotificationComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'angular4-notify-notification', [], null, null, null, View_NotificationComponent_0, exports.RenderType_NotificationComponent)), i0.ɵdid(1, 49152, null, 0, i1.NotificationComponent, [], null, null)], null, null);
}
exports.View_NotificationComponent_Host_0 = View_NotificationComponent_Host_0;
exports.NotificationComponentNgFactory = i0.ɵccf('angular4-notify-notification', i1.NotificationComponent, View_NotificationComponent_Host_0, { notification: 'notification' }, { dismissNotify: 'dismissNotify' }, []);
