// ngx-pagination/dist/ngx-pagination.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'ngx-pagination';
import * as i2 from '@angular/common';
export const NgxPaginationModuleNgFactory:i0.NgModuleFactory<i1.NgxPaginationModule> = i0.ɵcmf(i1.NgxPaginationModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(4608,i1.PaginationService,
          i1.PaginationService,([] as any[])),i0.ɵmpd(512,i2.CommonModule,i2.CommonModule,
          ([] as any[])),i0.ɵmpd(512,i1.NgxPaginationModule,i1.NgxPaginationModule,
          ([] as any[]))]);
    });
const styles_PaginationControlsComponent:any[] = ['\n.ngx-pagination {\n  margin-left: 0;\n  margin-bottom: 1rem; }\n  .ngx-pagination::before, .ngx-pagination::after {\n    content: \' \';\n    display: table; }\n  .ngx-pagination::after {\n    clear: both; }\n  .ngx-pagination li {\n    -moz-user-select: none;\n    -webkit-user-select: none;\n    -ms-user-select: none;\n    margin-right: 0.0625rem;\n    border-radius: 0; }\n  .ngx-pagination li {\n    display: inline-block; }\n  .ngx-pagination a,\n  .ngx-pagination button {\n    color: #0a0a0a; \n    display: block;\n    padding: 0.1875rem 0.625rem;\n    border-radius: 0; }\n    .ngx-pagination a:hover,\n    .ngx-pagination button:hover {\n      background: #e6e6e6; }\n  .ngx-pagination .current {\n    padding: 0.1875rem 0.625rem;\n    background: #2199e8;\n    color: #fefefe;\n    cursor: default; }\n  .ngx-pagination .disabled {\n    padding: 0.1875rem 0.625rem;\n    color: #cacaca;\n    cursor: default; } \n    .ngx-pagination .disabled:hover {\n      background: transparent; }\n  .ngx-pagination .ellipsis::after {\n    content: \'…\';\n    padding: 0.1875rem 0.625rem;\n    color: #0a0a0a; }\n  .ngx-pagination a, .ngx-pagination button {\n    cursor: pointer; }\n\n.ngx-pagination .pagination-previous a::before,\n.ngx-pagination .pagination-previous.disabled::before { \n  content: \'«\';\n  display: inline-block;\n  margin-right: 0.5rem; }\n\n.ngx-pagination .pagination-next a::after,\n.ngx-pagination .pagination-next.disabled::after {\n  content: \'»\';\n  display: inline-block;\n  margin-left: 0.5rem; }\n\n.ngx-pagination .show-for-sr {\n  position: absolute !important;\n  width: 1px;\n  height: 1px;\n  overflow: hidden;\n  clip: rect(0, 0, 0, 0); }'];
export const RenderType_PaginationControlsComponent:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_PaginationControlsComponent,data:{}});
function View_PaginationControlsComponent_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),4,'a',[['tabindex',
      '0']],[[1,'aria-label',0]],[[(null as any),'keyup.enter'],[(null as any),'click']],
      (_v,en,$event) => {
        var ad:boolean = true;
        if (('keyup.enter' === en)) {
          const pd_0:any = ((<any>i0.ɵnov((<any>(<any>(<any>_v.parent).parent).parent),
              2).previous()) !== false);
          ad = (pd_0 && ad);
        }
        if (('click' === en)) {
          const pd_1:any = ((<any>i0.ɵnov((<any>(<any>(<any>_v.parent).parent).parent),
              2).previous()) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted(1,(null as any),['\n                ',
      ' '])),(_l()(),i0.ɵeld(2,0,(null as any),(null as any),1,'span',[['class','show-for-sr']],
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted(3,(null as any),['',''])),(_l()(),i0.ɵted(-1,(null as any),['\n            ']))],
      (null as any),(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = ((_co.previousLabel + ' ') + _co.screenReaderPageLabel);
        _ck(_v,0,0,currVal_0);
        const currVal_1:any = _co.previousLabel;
        _ck(_v,1,0,currVal_1);
        const currVal_2:any = _co.screenReaderPageLabel;
        _ck(_v,3,0,currVal_2);
      });
}
function View_PaginationControlsComponent_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),4,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted(1,(null as any),['\n                ',' '])),(_l()(),i0.ɵeld(2,0,(null as any),
      (null as any),1,'span',[['class','show-for-sr']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted(3,(null as any),
      ['',''])),(_l()(),i0.ɵted(-1,(null as any),['\n            ']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.previousLabel;
        _ck(_v,1,0,currVal_0);
        const currVal_1:any = _co.screenReaderPageLabel;
        _ck(_v,3,0,currVal_1);
      });
}
function View_PaginationControlsComponent_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),7,'li',[['class',
      'pagination-previous']],[[2,'disabled',(null as any)]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted(-1,(null as any),[' \n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_3)),
      i0.ɵdid(3,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),['\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_4)),
      i0.ɵdid(6,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),['\n        ']))],
      (_ck,_v) => {
        const currVal_1:any = (1 < i0.ɵnov((<any>(<any>_v.parent).parent),2).getCurrent());
        _ck(_v,3,0,currVal_1);
        const currVal_2:any = i0.ɵnov((<any>(<any>_v.parent).parent),2).isFirstPage();
        _ck(_v,6,0,currVal_2);
      },(_ck,_v) => {
        const currVal_0:any = i0.ɵnov((<any>(<any>_v.parent).parent),2).isFirstPage();
        _ck(_v,0,0,currVal_0);
      });
}
function View_PaginationControlsComponent_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),7,'a',[['tabindex',
      '0']],(null as any),[[(null as any),'keyup.enter'],[(null as any),'click']],
      (_v,en,$event) => {
        var ad:boolean = true;
        if (('keyup.enter' === en)) {
          const pd_0:any = ((<any>i0.ɵnov((<any>(<any>(<any>_v.parent).parent).parent),
              2).setCurrent((<any>_v.parent).context.$implicit.value)) !== false);
          ad = (pd_0 && ad);
        }
        if (('click' === en)) {
          const pd_1:any = ((<any>i0.ɵnov((<any>(<any>(<any>_v.parent).parent).parent),
              2).setCurrent((<any>_v.parent).context.$implicit.value)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted(-1,(null as any),['\n                '])),
      (_l()(),i0.ɵeld(2,0,(null as any),(null as any),1,'span',[['class','show-for-sr']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted(3,(null as any),['',' '])),(_l()(),i0.ɵted(-1,(null as any),
          ['\n                '])),(_l()(),i0.ɵeld(5,0,(null as any),(null as any),
          1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted(6,(null as any),['',''])),(_l()(),i0.ɵted(-1,
          (null as any),['\n            ']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.screenReaderPageLabel;
    _ck(_v,3,0,currVal_0);
    const currVal_1:any = (<any>_v.parent).context.$implicit.label;
    _ck(_v,6,0,currVal_1);
  });
}
function View_PaginationControlsComponent_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),7,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted(-1,(null as any),['\n                '])),(_l()(),i0.ɵeld(2,
          0,(null as any),(null as any),1,'span',[['class','show-for-sr']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted(3,
          (null as any),['',' '])),(_l()(),i0.ɵted(-1,(null as any),['\n                '])),
      (_l()(),i0.ɵeld(5,0,(null as any),(null as any),1,'span',([] as any[]),(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted(6,
          (null as any),['',''])),(_l()(),i0.ɵted(-1,(null as any),[' \n            ']))],
      (null as any),(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.screenReaderCurrentLabel;
        _ck(_v,3,0,currVal_0);
        const currVal_1:any = (<any>_v.parent).context.$implicit.label;
        _ck(_v,6,0,currVal_1);
      });
}
function View_PaginationControlsComponent_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),7,'li',([] as any[]),
      [[2,'current',(null as any)]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted(-1,(null as any),['\n            '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_PaginationControlsComponent_6)),
      i0.ɵdid(3,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),['\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_7)),
      i0.ɵdid(6,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),['\n        ']))],
      (_ck,_v) => {
        const currVal_1:any = (i0.ɵnov((<any>(<any>_v.parent).parent),2).getCurrent() !== _v.context.$implicit.value);
        _ck(_v,3,0,currVal_1);
        const currVal_2:any = (i0.ɵnov((<any>(<any>_v.parent).parent),2).getCurrent() === _v.context.$implicit.value);
        _ck(_v,6,0,currVal_2);
      },(_ck,_v) => {
        const currVal_0:any = (i0.ɵnov((<any>(<any>_v.parent).parent),2).getCurrent() === _v.context.$implicit.value);
        _ck(_v,0,0,currVal_0);
      });
}
function View_PaginationControlsComponent_9(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),4,'a',[['tabindex',
      '0']],[[1,'aria-label',0]],[[(null as any),'keyup.enter'],[(null as any),'click']],
      (_v,en,$event) => {
        var ad:boolean = true;
        if (('keyup.enter' === en)) {
          const pd_0:any = ((<any>i0.ɵnov((<any>(<any>(<any>_v.parent).parent).parent),
              2).next()) !== false);
          ad = (pd_0 && ad);
        }
        if (('click' === en)) {
          const pd_1:any = ((<any>i0.ɵnov((<any>(<any>(<any>_v.parent).parent).parent),
              2).next()) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted(1,(null as any),['\n                 ',
      ' '])),(_l()(),i0.ɵeld(2,0,(null as any),(null as any),1,'span',[['class','show-for-sr']],
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted(3,(null as any),['',''])),(_l()(),i0.ɵted(-1,(null as any),['\n            ']))],
      (null as any),(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = ((_co.nextLabel + ' ') + _co.screenReaderPageLabel);
        _ck(_v,0,0,currVal_0);
        const currVal_1:any = _co.nextLabel;
        _ck(_v,1,0,currVal_1);
        const currVal_2:any = _co.screenReaderPageLabel;
        _ck(_v,3,0,currVal_2);
      });
}
function View_PaginationControlsComponent_10(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),4,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted(1,(null as any),['\n                 ',' '])),(_l()(),i0.ɵeld(2,0,(null as any),
      (null as any),1,'span',[['class','show-for-sr']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted(3,(null as any),
      ['',''])),(_l()(),i0.ɵted(-1,(null as any),['\n            ']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.nextLabel;
        _ck(_v,1,0,currVal_0);
        const currVal_1:any = _co.screenReaderPageLabel;
        _ck(_v,3,0,currVal_1);
      });
}
function View_PaginationControlsComponent_8(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),7,'li',[['class',
      'pagination-next']],[[2,'disabled',(null as any)]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted(-1,(null as any),['\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_9)),
      i0.ɵdid(3,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),['\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_10)),
      i0.ɵdid(6,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),['\n        ']))],
      (_ck,_v) => {
        const currVal_1:boolean = !i0.ɵnov((<any>(<any>_v.parent).parent),2).isLastPage();
        _ck(_v,3,0,currVal_1);
        const currVal_2:any = i0.ɵnov((<any>(<any>_v.parent).parent),2).isLastPage();
        _ck(_v,6,0,currVal_2);
      },(_ck,_v) => {
        const currVal_0:any = i0.ɵnov((<any>(<any>_v.parent).parent),2).isLastPage();
        _ck(_v,0,0,currVal_0);
      });
}
function View_PaginationControlsComponent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),10,'ul',[['class',
      'ngx-pagination'],['role','navigation']],[[1,'aria-label',0]],(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted(-1,(null as any),
      ['\n\n        '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_PaginationControlsComponent_2)),i0.ɵdid(3,16384,(null as any),0,i2.NgIf,
      [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
      i0.ɵted(-1,(null as any),['\n\n        '])),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_PaginationControlsComponent_5)),i0.ɵdid(6,
      802816,(null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
      {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),['\n\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_8)),
      i0.ɵdid(9,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),['\n\n    ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_1:any = _co.directionLinks;
        _ck(_v,3,0,currVal_1);
        const currVal_2:any = i0.ɵnov((<any>_v.parent),2).pages;
        _ck(_v,6,0,currVal_2);
        const currVal_3:any = _co.directionLinks;
        _ck(_v,9,0,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.screenReaderPaginationLabel;
        _ck(_v,0,0,currVal_0);
      });
}
export function View_PaginationControlsComponent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵted(-1,(null as any),['\n    '])),(_l()(),i0.ɵeld(1,
      0,(null as any),(null as any),5,'pagination-template',([] as any[]),(null as any),
      [[(null as any),'pageChange']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.PaginationControlsComponent = _v.component;
        if (('pageChange' === en)) {
          const pd_0:any = ((<any>_co.pageChange.emit($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(2,737280,[['p',4]],0,i1.PaginationControlsDirective,
      [i1.PaginationService,i0.ChangeDetectorRef],{id:[0,'id'],maxSize:[1,'maxSize']},
      {pageChange:'pageChange'}),(_l()(),i0.ɵted(-1,(null as any),['\n    '])),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_1)),
      i0.ɵdid(5,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(-1,(null as any),['\n    '])),
      (_l()(),i0.ɵted(-1,(null as any),['\n    ']))],(_ck,_v) => {
    var _co:i1.PaginationControlsComponent = _v.component;
    const currVal_0:any = _co.id;
    const currVal_1:any = _co.maxSize;
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:boolean = !(_co.autoHide && (i0.ɵnov(_v,2).pages.length <= 1));
    _ck(_v,5,0,currVal_2);
  },(null as any));
}
export function View_PaginationControlsComponent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'pagination-controls',
      ([] as any[]),(null as any),(null as any),(null as any),View_PaginationControlsComponent_0,
      RenderType_PaginationControlsComponent)),i0.ɵdid(1,49152,(null as any),0,i1.PaginationControlsComponent,
      ([] as any[]),(null as any),(null as any))],(null as any),(null as any));
}
export const PaginationControlsComponentNgFactory:i0.ComponentFactory<i1.PaginationControlsComponent> = i0.ɵccf('pagination-controls',
    i1.PaginationControlsComponent,View_PaginationControlsComponent_Host_0,{id:'id',
        maxSize:'maxSize',directionLinks:'directionLinks',autoHide:'autoHide',previousLabel:'previousLabel',
        nextLabel:'nextLabel',screenReaderPaginationLabel:'screenReaderPaginationLabel',
        screenReaderPageLabel:'screenReaderPageLabel',screenReaderCurrentLabel:'screenReaderCurrentLabel'},
    {pageChange:'pageChange'},([] as any[]));
